base/array.jl

```diff
@@ -383,6 +383,9 @@ function setindex!{T}(A::Array{T}, X::Array{T}, c::Colon)
     return A
 end
 
+setindex!{T}(A::Array{T}, x::Number, c::Colon) = fill!(A, x)
```

Comment:
Defining it on `setindex!` directly for all `AbstractArray`s leads to nasty ambiguity issues when custom arrays want to define their own behaviors.  They'll want to define a method with a more specific array type but less specific value and index types.

Interestingly, this is a new and exciting way for Base to break packages.